src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1426,6 +1465,8 @@ - (void)_handleEventReport:(NSArray<NSDictionary<NSString *, id> *> *)eventRepor
         BOOL isStartUpEvent = (eventPath.cluster.unsignedLongValue == MTRClusterIDTypeBasicInformationID)
             && (eventPath.event.unsignedLongValue == MTREventIDTypeClusterBasicInformationEventStartUpID);
         if (isStartUpEvent) {
+            // REVIEWERS:  this seems like a good place to set up / cache
+            // the essential device attributes - is it?
```

Comment:
This isn't really a great place to do it; ideally we would make this work along with the "device cache primed" bits, but those need some fixing first (on my plate to do ASAP).

That said, we could do this in `_handleSubscriptionEstablished` and `setPersistedClusterData`, which are two places we know our data is "best known at this time".... But again, if the device changes these values and reports new ones, shouldn't we update our concept of them?